kolibri/plugins/learn/assets/src/actions.js

```diff
@@ -57,21 +60,94 @@ function _collectionState(data) {
 }
 
 
+/*
+* Returns a promise that gets current channel.
+ */
+function _getCurrentChannel() {
+  let currentChannelId = null;
+  return new Promise(
+    (resolve, reject) => {
+      ChannelResource.getCollection({}).fetch()
+        .then((channelList) => {
+          const cookieCurrentChannelId = cookiejs.get('currentChannel');
+          if (channelList.some((channel) => channel.id === cookieCurrentChannelId)) {
+            currentChannelId = cookieCurrentChannelId;
+            resolve(currentChannelId);
+          } else {
+            currentChannelId = channelList[0].id;
+            resolve(currentChannelId);
+          }
+        })
+        .catch((error) => {
+        });
+    });
```

Comment:
minor style detail, but closing brackets/parens/etc should line up with the opening.

so in this case I would probably do

```javascript
function _getCurrentChannel() {
  let currentChannelId = null;
  return new Promise((resolve, reject) => {
    //...
  });
}
```